### PR TITLE
Add bundled GDB pretty-printers

### LINKED ISSUE #3357: Debugger displays incorrect values for strings
## Environment

* **Intellij-Rust plugin version:** v0.2.91.2115-183
* **Rust toolchain version:** 1.32.0 (9fda7c223 2019-01-16)
* **IDE name and version:** CLion 2018.3.4 Build #CL-183.5429.37
* **Operating system:** Ubuntu 18.04.1 LTS

## Problem description
Debugger displays incorrect values for strings in "Variables" window.

## Steps to reproduce
1. Create simple main.rs file with content
```
fn main() {
    let var1 = "123";
    let var2 = "456";
    println!("Hello, world!");
}
```
2. Set breakpoint to the last line

3. Click "Debug"

### PR DESCRIPTION
**Pull request by ortem**
Add bundled GDB pretty-printers.

Now it supports:
* struct, tuple
* `str`, `String`, `OsString` (with UTF-8 support)
* `Vec`, `VecDeque`
* `BTreeSet`, `BTreeMap` (thanks, @tromey!)
* `Rc`, `Arc`
* `Cell`, `Ref`, `RefMut`, `RefCell`

Fixes #3357 

There are some problems with enums because of wrong DWARF codes in rustc's codegen (it affects default rustc pretty-printers too); I'm going to fix it after the next rustc release.

screenshot 2018-11-19 at 11 54 56
screenshot 2018-11-19 at 11 54 16

screenshot 2018-11-27 at 19 04 44
screenshot 2018-11-27 at 19 00 47

screenshot 2018-11-29 at 18 56 55

### CODE REVIEW

**Comment by tromey** [comment]
> There are some problems with enums because of wrong DWARF codes in rustc's codegen

See https://github.com/rust-lang/rust/pull/54004.  If there's a bug, I definitely want to know about it -- but I suspect instead what is happening is that the new debuginfo is confusing the pretty-printers.  gdb 8.2 understands the new DWARF and consequently I disabled some of the default pretty-printers in https://github.com/rust-lang/rust/pull/55767.

**Comment by tromey** [comment]
... also I wonder if I need to fix https://sourceware.org/bugzilla/show_bug.cgi?id=20163, to allow varobjs to work better with rust enums.

**Comment by tromey** [comment]
> Maybe CLion (and probably other Rust IDEs) needs to support some new gdb features related to discriminated unions?

Assuming CLion is using MI varobjs (I didn't look so I don't know...) then it is that gdb bug I mentioned earlier -- basically varobj has to be taught a bit about discriminated unions.

**Comment by Undin** [comment]
@ortem Could you rebase onto master?

**Comment by ortem** [comment]
bors try

File: .travis.yml [hunk 1]
```diff
@@ -44,7 +44,7 @@ before_script:
 
 script:
   - travis_wait ./gradlew check -Pkotlin.incremental=false --no-daemon
-  - ./gradlew runPrettyPrintersTests --no-daemon
+  - if [ $RUST_VERSION == $CURRENT_RUST_VERSION ]; then ./gradlew runPrettyPrintersTests --no-daemon; fi
```
Comment (Undin) [inline .travis.yml]: Looks like we want to run pretty printer tests for all rust versions except old one
Let's replace condition with `$RUST_VERSION != $OLD_RUST_VERSION` (not sure about correct bash syntax)

**Comment by ortem** [comment]
bors retry

**Pull request merged**